Framework/Algorithms/src/GeneratePythonFitScript.cpp

```diff
@@ -224,6 +246,64 @@ std::string GeneratePythonFitScript::generateVariableSetupCode() const {
   return code;
 }
 
+std::string GeneratePythonFitScript::generateSequentialFitCode() const {
+  std::string code = "# Perform a sequential fit\n";
+  code += "output_workspaces, parameter_tables, normalised_matrices = [], [], []\n";
+  code += "for input_workspace, domain_data in input_data.items():\n";
+  code += "    fit_output = Fit(Function=function, InputWorkspace=input_workspace, WorkspaceIndex=domain_data[0],\n";
+  code += "                     StartX=domain_data[1], EndX=domain_data[2], MaxIterations=max_iterations,\n";
+  code += "                     Minimizer=minimizer, CostFunction=cost_function, EvaluationType=evaluation_type,\n";
+  code += "                     CreateOutput=True)\n";
+  code += "\n";
+  code += "    output_workspaces.append(fit_output.OutputWorkspace)\n";
+  code += "    parameter_tables.append(fit_output.OutputParameters)\n";
+  code += "    normalised_matrices.append(fit_output.OutputNormalisedCovarianceMatrix)\n";
+  code += "\n";
+  code += "    # Use the parameters in the previous function as the start parameters of the next fit\n";
+  code += "    function = fit_output.Function\n";
+  code += "\n";
+  code += "# Group the output workspaces from the sequential fit\n";
+  code += "GroupWorkspaces(InputWorkspaces=output_workspaces, OutputWorkspace=\"Sequential_Fit_Workspaces\")\n";
+  code += "GroupWorkspaces(InputWorkspaces=parameter_tables, OutputWorkspace=\"Sequential_Fit_Parameters\")\n";
+  code += "GroupWorkspaces(InputWorkspaces=normalised_matrices, ";
+  code += "OutputWorkspace=\"Sequential_Fit_NormalisedCovarianceMatrices\")\n\n";
+  return code;
+}
+
+std::string GeneratePythonFitScript::generateSimultaneousFitCode() const {
+  std::string code = "# Perform a simultaneous fit\n";
+  code += "input_workspaces = list(input_data.keys())\n";
+  code += "domain_data = list(input_data.values())\n\n";
+  code += "fit_output = Fit(Function=function,\n";
+
+  code += "                 InputWorkspace=input_workspaces[0], WorkspaceIndex=domain_data[0][0], "
+          "StartX=domain_data[0][1], EndX=domain_data[0][2],\n";
+
+  std::vector<std::string> const inputWorkspaces = getProperty("InputWorkspaces");
+  for (auto i = 1u; i < inputWorkspaces.size(); ++i) {
```

Comment:
For this I imagine the Python file looking like:
```python 
# Double braces tend do me de-facto for Python templating (see Jinja)
InputWorkspace_{{i}}=input_workspaces[{{i}}], WorkspaceIndex_{{i}}=domain_data[{{i}}][0],`
```

Then it's a simple method to do string substitution using [this method](https://stackoverflow.com/a/2548212):
```c++
std::string prepareTemplate(){
   std::string code = loadFromFile("template.py");
   for (auto i = 1; i < inputWorkspaces.size(); i++){
       std::string snippet = loadFromFile("template_ws_index.py");
       // Constants omitted for brevity
       replaceAll(snippet, "{{i}}", i);
       code += snippet
    }
    return code;
}
```